syntax = "proto3";

package remote_shell;

service RemoteShell {
    // Client initiates connection and maintains stream for receiving commands
    rpc ConnectClient (ClientInfo) returns (stream Command) {}
    
    // Client sends command results back to server
    rpc SendCommandResult (CommandResult) returns (CommandAck) {}
}

message ClientInfo {
    string hostname = 1;
    string os_info = 2;
    string username = 3;
    string client_id = 4;
}

message Command {
    string command_id = 1;
    string command = 2;
    int32 timeout = 3;
}

message CommandResult {
    string command_id = 1;
    string output = 2;
    int32 exit_code = 3;
    string error = 4;
    string client_id = 5;
}

message CommandAck {
    bool received = 1;
}
